src/query/plan/preprocess.cpp

```diff
@@ -184,7 +186,8 @@ void AddExpansionsToMatching(std::vector<Expansion> &expansions, Matching &match
   }
 }
 
-auto SplitExpressionOnAnd(Expression *expression) {
+template <typename F1, typename F2>
+void SplitExpressionOnAnd(Expression *expression, F1 &&if_cb, F2 &&else_cb) {
```

Comment:
Only two usages...inline into both places and refactor them to have simple implementations. The generic callbacks reduce readability.

Reply:
Sure